client/src/Components/ImageSlider/ImageSlider.elm

```diff
@@ -20,59 +44,192 @@ update shared msg model =
 
         NextImage ->
             let
-                currentI =
-                    List.elemIndex model.currentImage.original (List.map .original model.images)
-            in
-            ( { model
-                | currentImage =
-                    case currentI of
-                        Just i ->
-                            if i + 1 >= List.length model.images then
-                                Maybe.withDefault model.currentImage <| List.getAt 0 model.images
+                i =
+                    model.currentImage
 
-                            else
-                                Maybe.withDefault model.currentImage <| List.getAt (i + 1) model.images
+                ( newCurrentImage, cmd ) =
+                    if i + 1 >= Array.length model.images then
```

Comment:
Perhaps `newI = modBy (Array.length model.images) (i + 1)` instead of `if`?

And `newI = modBy (Array.length model.images) (i - 1)` slightly below

Reply:
My brain is too small to understand why this works, but it works 😅